Code/Framework/AzCore/AzCore/RTTI/TypeInfo.h

```diff
@@ -523,6 +518,91 @@ namespace AZStd
     AZ_TYPE_INFO_INTERNAL_SPECIALIZED_TEMPLATE_BOTHFIX_UUID_DECL(AZStd::basic_fixed_string, AZ_TYPE_INFO_INTERNAL_TYPENAME, AZ_TYPE_INFO_INTERNAL_AUTO, AZ_TYPE_INFO_INTERNAL_TYPENAME);
 }
 
+namespace AZStd
+{
+    // GetO3deTypeName/GetO3deTypeId overload for AZStd::span<T, Extent>
+    // Note the type ID only takes the type T template parameter into account, not the Extent template parameter.
+    // An `AZStd::span<AZ::Component*, 50>` and `AZStd::span<AZ::Component*, 100>` will have the same type ID, as the second template argument is not aggregated to the AZStd::span template ID.
+    inline constexpr AZ::TemplateId GetO3deTemplateId(AZ::Adl, AZ::AzGenericTypeInfo::Internal::TemplateIdentityTypeAuto<AZStd::span>)
+    {
+        constexpr AZ::TypeId prefixUuid{ "{2FCDBAB3-45E0-4159-A91D-FD1D37056C0F}" };
+        constexpr AZ::TypeId postfixUuid{};
+        if constexpr (!prefixUuid.IsNull())
+        {
+            return prefixUuid;
+        }
+        else if constexpr (!postfixUuid.IsNull())
+        {
+            return postfixUuid;
+        }
+        else
+        {
+            return AZ::TemplateId{};
+        }
+    }
+    template<typename T1>
+    AZ::TypeNameString GetO3deTypeName(AZ::Adl, AZStd::type_identity<AZStd::span<T1>>)
```

Comment:
not `constexpr`?

Reply:
The `AZ::Internal::GetTypeName` that is used in this function is not constexpr and therefore this function cannot be marked as such.